packages/@extension-kid/toast-notifications/__tests__/ToastExtension-test.ts

```diff
@@ -0,0 +1,190 @@
+      ext.Toast$.subscribe(nextCallback);
+      nextCallback.mockReset();
+
+      ext.push(notification01);
+
+      expect(nextCallback).toHaveBeenCalledTimes(1);
+      expect(nextCallback).toHaveBeenCalledWith([notification01]);
+      nextCallback.mockReset();
+
+      ext.push(notification02);
+      expect(nextCallback).toHaveBeenCalledTimes(1);
+      expect(nextCallback).toHaveBeenCalledWith([
+        notification01,
+        notification02
+      ]);
+    });
+  });
+
+  describe("#dismissToast", () => {
+    it("removes notifications when dismissed", () => {
+      const ext = new ToasterExtension();
+      const notification01 = new ToastNotification("unit test one");
+      const notification02 = new ToastNotification("unit test two");
+      const nextCallback = jest.fn();
+      ext.Toast$.subscribe(nextCallback);
+      nextCallback.mockReset();
+
+      ext.push(notification01);
+      ext.push(notification02);
+      nextCallback.mockReset();
+
+      ext.dismissToast(notification01.id);
+      expect(nextCallback).toHaveBeenCalledTimes(1);
+      expect(nextCallback).toHaveBeenCalledWith([notification02]);
+    });
+
+    it("doesn't emit if notification not found", () => {
+      const ext = new ToasterExtension();
+      const nextCallback = jest.fn();
+      ext.Toast$.subscribe(nextCallback);
+      expect(nextCallback).toHaveBeenCalledWith([]);
+      nextCallback.mockReset();
+
+      ext.dismissToast("I.dont.exist");
+      expect(nextCallback).not.toHaveBeenCalled();
+    });
+
+    it("invokes notification callback when dismissed", () => {
+      const ext = new ToasterExtension();
+      const notificationCallback = jest.fn();
+
+      const notification01 = new ToastNotification(
+        "unit test one",
+        undefined,
+        notificationCallback
+      );
+      ext.push(notification01);
+      ext.dismissToast(notification01.id);
+      expect(notificationCallback).toHaveBeenCalledTimes(1);
+      expect(notificationCallback).toHaveBeenCalledWith(
+        ToastCallbackType.Dismiss,
+        notification01
+      );
+    });
+
+    it("invokes notification callback when dismissed", () => {
+      const ext = new ToasterExtension();
+      const notificationCallback = jest.fn();
+
+      const notification01 = new ToastNotification(
+        "unit test one",
+        undefined,
+        notificationCallback
+      );
+      ext.push(notification01);
+      ext.dismissToast(notification01.id);
+      expect(notificationCallback).toHaveBeenCalled();
+    });
+  });
+
+  describe("#toastPrimaryAction", () => {
+    it("removes notification action executed", () => {
+      const ext = new ToasterExtension();
+      const notification01 = new ToastNotification("unit test one");
+      const nextCallback = jest.fn();
+      ext.Toast$.subscribe(nextCallback);
+      ext.push(notification01);
+      nextCallback.mockReset();
+      expect(ext.Toast$.getValue()).toEqual([notification01]);
```

Comment:
i think expecting this here is redundant (we already tested that above, right?) and may also a little brittle as it relies on impl-details (`getValue`).